include/range_mapper.h

```diff
@@ -189,27 +175,63 @@ namespace access {
 		}
 	};
 
+	/// Optional parameter to the `neighborhood` constructor to specify in what shape the accessed region should extend from the work item.
+	enum neighborhood_shape {
+		along_axes,   ///< The neighborhood extends along each axis separately, but not diagonally (in 2D, into a "+" shape).
+		bounding_box, ///< The neighborhood extends along in all dimensions simultaneously to produce a single bounding box.
+	};
+
+	/// Declares a buffer access that extends from the current work item by a symmetric boundary offset, either in all directions in the shape of a bounding
+	/// box (default), or along each axis separately (without "diagonal" boundary elements). Buffer and kernel dimensions must both match the `Dims` parameter.
+	///
+	/// This is typically used in stencil applications. For bounding-box neighborhoods,
+	/// - `neighborhood(1)` declares the read for a 1D 3-point stencil,
+	/// - `neighborhood(1, 1)` for a 2D 9-point stencil, and
+	/// - `neighborhood(1, 1, 1)` for a 3D 27-point stencil.
+	///
+	/// For neighborhoods defined along axes only,
+	/// - `neighborhood(range(1), along_axes)` declares the read for a 1D 3-point stencil,
+	/// - `neighborhood(range(1, 1), along_axes)` for a 2D 5-point stencil, and
+	/// - `neighborhood(range(1, 1, 1), along_axes)` for a 3D 7-point stencil.
```

Comment:
Hm, what's the reasoning for adding a new `range` overload instead of adding the default argument to the existing constructors?

Reply:
The existing ones are now deprecated to notify our users that they should look at the optional shape parameter for their code, which might have used the bounding-box neighborhood before without intending it.